src/core/expr/py_join.cc

```diff
@@ -62,7 +62,51 @@ except: ValueError
 
 See Also
 --------
+
 - `Tutorial on joins <../../start/quick-start.html#join>`_
+
+Examples
+--------
+
+.. code-block:: python
+
+    df1 = dt.Frame("""date	       X1	X2
+            	01-01-2020	H	10
+            	01-02-2020	H	30
+            	01-03-2020	Y	15
+            	01-04-2020	Y	20""")
+
+    df2 = dt.Frame("""X1	X3
+            	   H	5
+                       Y	10""")
+
+
+First, create a key on the right frame (``df2``). Note that the join key(``X1``) has unique values and has the same name in the left frame(``df1``)::
+
+    df2.key = "X1"
+
+Join is now possible::
+
+    df1[:, :, join(df2)]
+
+        date	        X1	X2	X3
+    0	01-01-2020	H	10	5
+    1	01-02-2020	H	30	5
+    2	01-03-2020	Y	15	10
+    3	01-04-2020	Y	20	10
+
+You can access the values in the right frame in an expression, using the secondary namespace object `g  <https://datatable.readthedocs.io/en/latest/api/dt/g.html>`__::
```

Comment:
```suggestion
You can refer to columns of the joined frame using prefix :data:`g. <dt.g>`, similar to how columns of the left frame can be accessed using prefix :data:`f. <dt.f>`.
```

Reply:
Please note that internal links within the documentation should use dedicated syntax; in this case the link  to `g.` is 
```
:data:`g. <dt.g>`
```
This has several  advantages:
- the link is not rendered  as external (with a tiny arrow at the end);
- the link works properly when rendered locally;
- the link will work correctly with different versions of the documentation (i.e. v0.11, v1.0, latest, etc).